packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -328,29 +220,170 @@ export default function BlockList( {
 		);
 	};
 
+	const { blockToolbar, headerToolbar, floatingToolbar } = styles;
+
+	const containerStyle = {
+		flex: isRootList ? 1 : 0,
+		// We set negative margin in the parent to remove the edge spacing between parent block and child block in ineer blocks.
+		marginVertical: isRootList ? 0 : -marginVertical,
+		marginHorizontal: isRootList ? 0 : -marginHorizontal,
+	};
+
+	const isContentStretch = contentResizeMode === 'stretch';
+	const isMultiBlocks = blockClientIds.length > 1;
+	const { isWider } = alignmentHelpers;
+	const extraScrollHeight =
+		headerToolbar.height +
+		blockToolbar.height +
+		( isFloatingToolbarVisible ? floatingToolbar.height : 0 );
+
+	const scrollViewStyle = [
+		{ flex: isRootList ? 1 : 0 },
+		! isRootList && styles.overflowVisible,
+	];
+
 	// Use of Context to propagate the main scroll ref to its children e.g InnerBlocks.
-	const blockList = isRootList ? (
-		<BlockListProvider
-			value={ {
-				...DEFAULT_BLOCK_LIST_CONTEXT,
-				scrollRef: scrollViewRef.current,
-			} }
+	return (
+		<View
+			style={ containerStyle }
+			onAccessibilityEscape={ clearSelectedBlock }
+			onLayout={ onLayout }
+			testID="block-list-wrapper"
 		>
-			<BlockDraggableWrapper isRTL={ isRTL }>
-				{ ( { onScroll } ) => renderList( { onScroll } ) }
-			</BlockDraggableWrapper>
-		</BlockListProvider>
-	) : (
-		<BlockListConsumer>
-			{ ( { scrollRef } ) =>
-				renderList( {
-					parentScrollRef: scrollRef,
```

Comment:
You're correct, since we are changing the way we render InnerBlocks, we won't need to do this anymore and we'll only need to set the main list ref `scrollRef` so it can be used by the Drag&Drop feature.

As it looks now, you took the right approach to remove the usage of `BlockListConsumer`.